src/sled_store.rs

```diff
@@ -36,16 +36,25 @@ pub enum SetError {
 }
 
 impl SledStore {
-    pub(crate) fn new(config: &StoreConfig) -> Self {
-        let dirs = ProjectDirs::from(
-            config.qualifier.as_deref().unwrap_or(""),
-            &config.organization,
-            &config.application,
-        );
-        let parent_dir = match dirs.as_ref() {
+    pub(crate) fn new(config: Option<&StoreConfig>, custom_path: Option<&Path>) -> Self {
+        let dirs = match config {
+            Some(config) => ProjectDirs::from(
+                config.qualifier.as_deref().unwrap_or(""),
+                &config.organization,
+                &config.application,
+            ),
+            None => None,
+        };
```

Comment:
We now have quite a bit of duplicated path-logic, right? Could you maybe move this to a method on the enum i suggested? Gated behind a cfg attribute?

Reply:
I'll get it done :)